### ISSUE TITLE
Enable Cython 3

### ISSUE THREAD

**Issue by andrebienemann** [stefan-jansen/bcolz-zipline#40]
Was going to commit a fix that enables Cython 3, but realised that I have no permissions to push to this repository. Hence, opening a ticket to fix line `1685` in `bcolz/carray_ext.pyx` with `rsize = isize // newlen`. Floor division solves the build issues that currently anyone would experience with Cython >= 3.0.0

**Comment by gnzsnz** [comment]
hi, this is awesome.

do you have this in a public repository? i would love to test it locally

I would suggest that you submit a pull request for this.

**Comment by gnzsnz** [comment]
@andrebienemann thanks for your hint on fixing this issue.

based on your description of the I applied the [change](https://github.com/gnzsnz/bcolz-zipline/blob/17f06292b1ea0773e76aec5fb12d725d34a3beb2/bcolz/carray_ext.pyx#L1685) and manage to test it.

I can not reproduce this issue in linux/amd64, only on linux/arm64 I get the error.

linux/amd64 can install and compile bcolz-zipline with cython3 with no problems. if i install numexpr i get a warning while running the tests. but it completes all tests. your patch is not needed based on my tests for linux/amd64

on linux/arm64 i can only install bcolz-zipline if i apply your patch. otherwise fails during instalation. after applying your patch i can successfully run tests. if i install numexpr i get a warning, same warning that for linux/amd64, but all tests complete successfully.

this works fine for me! thanks!

**Comment by gnzsnz** [comment]
I would like to amend my previous comment.

I did additional tests and realized that the fix on line 1685 is not actually fixing the issue. the fix is comming from PR #37 
If i install from [github](pip install git+https://github.com/gnzsnz/bcolz-zipline.git@main) rather than pypi everything works fine, there is no need to fix line 1685

**Comment by andrebienemann** [comment]
Hey, thank you for your comments!

I should've described the issue in more detail.

As mentioned above, the installation of bcolz-zipline only fails on linux/arm64. When building wheels on this platform, the division in line `1685` of the `bcolz/carray_ext.pyx` file causes an error `Cannot assign type 'double' to 'npy_intp'`. It happens because Cython 3 is not comfortable assigning C double to C integer. The solution is to use floor devision which produces an integer as the result of the operation. One way to prevent this issue is to use Cython below version 3 by adding a build constraint as it's been achived through the PR [#37](https://github.com/stefan-jansen/bcolz-zipline/pull/37). However, the long term solutions would argaubly be to fix the devision operation and therefore enable newer versions of Cython.

I tested the fix with the source code from 977ce34, before [#37](https://github.com/stefan-jansen/bcolz-zipline/pull/37) was merged, and I was able to suceffully build the library. Forking this reposiotry and creating a PR back to fix one line of code seems a bit too much of overhaed to me. Hence, leaving this fix to the maintainers.
